src/compositions/filters.ts

```diff
@@ -559,6 +559,76 @@ export function useDeploymentsFilter(defaultValue: MaybeReactive<DeploymentsFilt
   return useUnionFilter<DeploymentsFilter>(defaultValue, defaultDeploymentSort)
 }
 
+function usePaginationUnionFilter<T extends PaginationUnionFilter>(defaultValue: MaybeReactive<T>, defaultSort: Exclude<T['sort'], undefined>): UseFilter<WithPage<T>> {
```

Comment:
Worth pointing out the `WithPage` part of the return type. Because `p-pager` complains about `page` being `number | undefined` I opted to have these compositions accept the normal filter (where page is optional) but always return `number` for page and default the value to `1`. This matches the api logic and makes the types for these filters much easier to work with at implementation.

Reply:
Also, this composition is a straight copy/paste of the `useUnionFilter` composition and just changed the types and added the page default. So probably not worth digging into how this actually works unless you're just curious.